src/repositories/recurringDonationRepository.ts

```diff
@@ -29,14 +32,35 @@ export const createNewRecurringDonation = async (params: {
 };
 export const updateRecurringDonation = async (params: {
   recurringDonation: RecurringDonation;
-  txHash: string;
-  flowRate: string;
-  anonymous: boolean;
+  txHash?: string;
+  flowRate?: string;
+  anonymous?: boolean;
+  status?: string;
 }): Promise<RecurringDonation> => {
-  const { recurringDonation, txHash, anonymous, flowRate } = params;
-  recurringDonation.txHash = txHash;
-  recurringDonation.flowRate = flowRate;
-  recurringDonation.anonymous = anonymous;
+  const { recurringDonation, txHash, anonymous, flowRate, status } = params;
+  if (txHash && flowRate) {
+    recurringDonation.txHash = txHash;
+    recurringDonation.flowRate = flowRate;
+  }
+
+  if (anonymous) {
+    recurringDonation.anonymous = anonymous;
+  }
+
+  if (
+    recurringDonation.status === RECURRING_DONATION_STATUS.ACTIVE &&
+    status === RECURRING_DONATION_STATUS.ENDED
+  ) {
+    recurringDonation.status = status;
+  }
+
+  if (
+    recurringDonation.status === RECURRING_DONATION_STATUS.ENDED &&
+    status === RECURRING_DONATION_STATUS.ARCHIVED
+  ) {
+    recurringDonation.status = status;
+  }
+
```

Comment:
in other cases representing "else" conditions of each of above scenarios, it would be helpful to log why we didnt update status

Reply:
Thanks! Ill take it into account